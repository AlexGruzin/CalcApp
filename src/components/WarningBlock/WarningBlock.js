import React from 'react';
import PropTypes from 'prop-types';
import { useHistory } from 'react-router-dom';
import Button from 'components/Button';

import './WarningBlock.styles.less';

const WarningBlock = ({ text, buttonText, route, textBeforeButton }) => {
  const history = useHistory();

  return (
    <div className="warning-block">
      <div className="warning-block__info">
        <img className="warning-block__info-svg" src="/svg/warning.svg" alt="info-svg" />
        {text}
      </div>
      <div className="warning-block__btn">
        {textBeforeButton}
        {buttonText && (
          <Button className="custom-button_blue" onClick={() => history.push(route)}>
            {buttonText}
          </Button>
        )}
      </div>
    </div>
  );
};

WarningBlock.propTypes = {
  text: PropTypes.string.isRequired,
  buttonText: PropTypes.string,
  route: PropTypes.string,
  textBeforeButton: PropTypes.string,
};

WarningBlock.defaultProps = {
  buttonText: '',
  route: '',
  textBeforeButton: '',
};

export default WarningBlock;
