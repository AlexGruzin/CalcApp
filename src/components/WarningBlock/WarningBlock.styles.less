@import "~styles/generic.less";

.warning-block {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: auto;
  padding: var(--space-x3);
  border-radius: 15px;
  background: var(--white);
  font-size: var(--fontSize-superTiny);

  & > div {
    display: flex;
    align-items: center;
  }

  &__info-svg {
    margin-right: var(--space-x2);
  }

  &__btn button {
    margin-left: var(--space-x3);
    font-size: var(--fontSize-ultraTiny);
    border-radius: 15px;
  }
}
