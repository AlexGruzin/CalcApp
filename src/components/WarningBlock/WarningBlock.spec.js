import React from 'react';
import { render } from '@testing-library/react';
import WarningBlock from './WarningBlock';

describe('WarningBlock tests', () => {
  it('Check snapshots of WarningBlock are equal.', () => {
    const { asFragment } = render(<WarningBlock text="" buttonText="" route="" textBeforeButton="" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
