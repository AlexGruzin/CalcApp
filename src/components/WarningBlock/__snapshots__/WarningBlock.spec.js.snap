// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WarningBlock tests Check snapshots of WarningBlock are equal. 1`] = `
<DocumentFragment>
  <div
    class="warning-block"
  >
    <div
      class="warning-block__info"
    >
      <img
        alt="info-svg"
        class="warning-block__info-svg"
        src="/svg/warning.svg"
      />
    </div>
    <div
      class="warning-block__btn"
    />
  </div>
</DocumentFragment>
`;
